tests/shared/core/test_domain.py

```diff
@@ -2067,3 +2067,112 @@ def test_merge_domain_with_separate_session_config():
         domain.session_config.session_expiration_time
         == expected_session_expiration_time
     )
+
+
+@pytest.mark.parametrize(
+    "actions, expected_result",
+    [
+        (
+            [
+                {"action_hello_world": {"send_domain": False}},
+                {"action_say_something": {"send_domain": True}},
+                {"action_calculate": {"send_domain": True}},
+                "action_no_domain",
+            ],
+            ["action_say_something", "action_calculate"],
+        )
+    ],
+)
+def test_collect_actions_which_explicitly_need_domain(
```

Comment:
Could we add more edge cases here (when actions list is empty, when actions list contains only strings etc)